packages/sit-onyx/src/components/OnyxCheckboxGroup/OnyxCheckboxGroup.vue

```diff
@@ -0,0 +1,102 @@
+<script lang="ts" setup generic="TValue extends string | number | boolean">
+import { injectI18n } from "@/i18n";
+import { OnyxHeadline, type OnyxCheckboxProps } from "@/index";
+import { computed } from "vue";
+import OnyxCheckbox from "../OnyxCheckbox/OnyxCheckbox.vue";
+import type { OnyxCheckboxGroupProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxCheckboxGroupProps<TValue>>(), {
+  modelValue: () => [],
+  direction: "vertical",
+  showCheckAll: false,
+  disabled: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the checked checkboxes change.
+   */
+  "update:modelValue": [value: TValue[]];
+}>();
+
+const { t } = injectI18n();
+
+const handleUpdate = (id: TValue, isChecked: boolean) => {
+  const newValue = isChecked ? [...props.modelValue, id] : props.modelValue.filter((i) => i !== id);
+  emit("update:modelValue", newValue);
+};
+
+const enabledOptions = computed(() => props.options.filter((i) => !i.disabled));
+
+const handleMasterCheckboxChange = (isChecked: boolean) => {
+  const newValue = isChecked ? enabledOptions.value.map(({ id }) => id) : [];
+  emit("update:modelValue", newValue);
+};
+
+/**
+ * Current master checkbox state.
+ * - checked if all options are checked
+ * - indeterminate if at least one but not all options are checked
+ * - unchecked if no options are checked
+ */
+const masterCheckboxState = computed<Partial<OnyxCheckboxProps>>(() => {
+  const availableOptionIds = enabledOptions.value.map(({ id }) => id);
+  const currentValues = props.modelValue.filter((i) => availableOptionIds.includes(i));
+
+  if (!availableOptionIds.length || !currentValues.length) return { modelValue: false };
+  if (currentValues.length === availableOptionIds.length) return { modelValue: true };
+  return { indeterminate: true };
```

Comment:
According to specs indeterminate is equivalent to false.
```suggestion
  return { indeterminate: true, modelValue: false };
```